### PR TITLE
Fixing shellcheck warnings in gitlib.sh

### PR DESCRIPTION
**Pull request by pswica**
Please see: https://github.com/kubernetes/release/pull/742 and https://github.com/kubernetes/release/issues/726 for more details and context.

Let me know anything else to improve on :)

### CODE REVIEW

File: lib/gitlib.sh [hunk 1]
```diff
@@ -189,7 +191,7 @@ gitlib::current_branch () {
 
   [[ -n "$repo_dir" ]] && git_args=("-C" "$repo_dir")
 
-  if ! git ${git_args[*]} rev-parse --abbrev-ref HEAD 2>/dev/null; then
+  if ! git "${git_args[*]}" rev-parse --abbrev-ref HEAD 2>/dev/null; then
```
Comment (spiffxp) [inline lib/gitlib.sh]: We want each element of the `git_args` array to be treated as its own shell word; `[*]` will expand to only one word
```suggestion
  if ! git "${git_args[@]}" rev-parse --abbrev-ref HEAD 2>/dev/null; then
```

Comment (pswica) [reply lib/gitlib.sh]: Thanks a lot for your review. I made the change

**Pull request merged**